### PR TITLE
CNV-23983: Create a modal form to add a boot volume

### PR DESCRIPTION
**Pull request by avivtur**
Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

## 📝 Description

A modal to create a DataVolume (with upload file or cloning an existing PVC) that adds the preferences and instanceType to a PVC as labels

## 🎥 Demo

https://user-images.githubusercontent.com/67270715/211275270-bd7bc47f-8794-4077-acc5-33810b39a569.mp4

### To-Do
- [ ] Verify backend delegated instance type and preference labels from DataVolume to PVC
- [ ] Verify with backend if resources created as demo shows are correct

### CODE REVIEW

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/constants.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export enum INSTANCE_TYPES_SECTIONS {
+  SELECT_VOLUME = 1,
+  SELECT_INSTANCE_TYPE = 2,
+  VM_DETAILS = 3,
+}
+
+export const VirtualMachineClusterPreferenceGroupVersionKind = {
```
Comment (pcbailey) [inline src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/constants.ts:7]: Shouldn't this and the one below go in kubevirt-api? Are these just temporary until you get them added there?

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+export const getName = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
```
Comment (pcbailey) [inline src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/utils.ts:3]: We already have a function just like this in cdi-uploader. We should probably move it somewhere in @kubevirt-utils.

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/components/AddBootableVolumeButton/components/AddBootableVolumeModal/components/FilterSelect/utils/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React, { ReactElement } from 'react';
+
+import { SelectOption } from '@patternfly/react-core';
+
+export const getSelectOption = (optionLabel: string) => (name: string) =>
```
Comment (metalice) [inline src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/components/AddBootableVolumeButton/components/AddBootableVolumeModal/components/FilterSelect/utils/utils.tsx:5]: hmm... get*** is misleading in this case, as this function eventually returns a component.
util files usually don't return elements/components.

**Comment by metalice** [comment]
very nice! GJ! left a small comment about 2 functions in the util file

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/constants.ts (hunk 1 above)
Comment (avivtur) [reply src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/constants.ts:10]: I added the proper models to kubevirt-api, and changed it

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/utils.ts (hunk 2 above)
Comment (avivtur) [reply src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/utils.ts:3]: you're right, I've changed it to utils/resources/shared.ts

File: src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/hooks/useInstanceTypesAndPreferences.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    K8sResourceCommon[]
+  >({
+    groupVersionKind: VirtualMachineClusterInstancetypeGroupVersionKind,
+    isList: true,
+  });
+
+  const loaded = preferencesLoaded && instanceTypesLoaded;
+  const loadError = preferencesLoadError || instanceTypesLoadError;
+
+  if (!loaded || loadError || isEmpty(preferences) || isEmpty(instanceTypes)) {
```
[the remark this answers is not in the document]
Comment (avivtur) [reply src/views/catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/hooks/useInstanceTypesAndPreferences.ts:38]: changed

**Pull request merged**